packages/zk-transfer-web/src/libs/types.ts

```diff
@@ -27,27 +33,17 @@ export interface ZkToken extends Map<string, any> {
   abi: []; // Some constants
 }
 
-export type Token = ZkToken | ERCToken;
+export interface ERC20Token extends ZkToken {}
 
-export interface TokenList<T> extends ArrayLike<T> {
-  values(): Object;
-  map<U>(f: (first: T, key: string) => U): any;
-  forEach<U>(f: (first: T, key: string) => U): void;
-  filter<U>(f: (first: T, key: string) => U): TokenList<T>;
-  mapEntries<U>(f: (entry: KeyValuePair<T>, key: string) => U): TokenList<T>;
-}
+export type Token = ZkToken | ERC20Token;
 
-export type KeyValuePair<L> = [string, L];
+export type NoteList = List<Note>;
 
-export interface NoteList extends TokenList<Note> {
-  get(address: string): NoteList;
-  filter<U>(f: (first: Note, key: string) => U): NoteList;
-  mapEntries<U>(f: (first: KeyValuePair<Note>, key: string) => U): NoteList;
-}
+export type TokenMap = Map<TokenAddress, Token>;
 
-export interface BN {
-  add(address: string): Number;
-}
+export type TokenNotesPair = [TokenAddress, NoteList];
+
+export type TokenNotesMap = Map<TokenAddress, NoteList>;
```

Comment:
I'd prefer this type to be `TokenAddressToNoteList` as it tells you exactly
what the types and meaning of the keys and values are.